mbed_project/mbed_program.py

```diff
@@ -25,27 +25,25 @@
 class MbedProgram:
     """Represents an Mbed program.
 
-    An `MbedProgram` consists of a git repo and `MbedProgramData`. The `MbedProgramData` contains configuration files
-    used for building an Mbed application, and a .mbed file which defines the program's root path.
-
-    `MbedProgram` provides classmethods to cope with different initialisation scenarios.
+    An `MbedProgram` consists of a git repo, a copy of `MbedOS`, a set of `MbedProgramFiles` and a collection of
```

Comment:
I think this is ok, but just to be super pedantic - couldn't Mbed OS be considered one of the external libraries since it is referenced by a .lib file? Would it be an invalid program if you hadn't downloaded the copy of Mbed OS?

Reply:
Yeah I thought about that too. I ended up coming to the conclusion that as `mbed-os.lib` is required to define a program I should treat the existence of it as a special case. The reason we have the `MbedOS` object is so we can reference paths in the Mbed OS source tree. Open to suggestions here.

Reply:
How about `a copy of or reference to 'MbedOS',`?

Reply:
Sounds good, I'll add that.

Reply:
acea7941390bd951d7199e61b51d9f5eb63da11b